src/Main.java

```diff
@@ -2251,11 +2259,20 @@ else if (region == 0x12) {
 			&& (difficulty > Difficulty.EASY || inventory.contains(Items.JUMP_BOOTS));
             }
             else if (region == 0x15) {
+				// New MINOR GLITCHES Logic for Inside 2nd Hill.
+				// You can skip needing Boots if you have Red Overalls by breaking the blocks in a specific way, and then doing a walljump after obtaining this check.
+				// This can only be in logic on MINOR GLITCHES if it specifically contains the Red Key.
                 return inventory.contains(Items.GARLIC) && canLift(inventory)
-                        && (canSwim(inventory) || (difficulty >= Difficulty.HARD && inventory.contains(Items.JUMP_BOOTS)));
+                        && (canSwim(inventory) 
+							|| ((difficulty >= Difficulty.HARD && inventory.contains(Items.JUMP_BOOTS)) 
+							|| (difficulty >= Difficulty.S_HARD && canSuperGP(inventory)));
```

Comment:
This needs one more `)` before the semicolon.

Separately, why would this trick only be in logic if it's the red key? Is it impossible to do the same setup at the first hill to also get back to the gray chest (and the blue chest, if you have the pump)? (I'm honestly asking, I haven't tried this myself yet.) **If** doing this trick truly locks you out of getting to the gray and blue chests, this is a situation where we would need to add a `keyColor == 1` check to this route.

Reply:
`keyColor == 1` was originally in this because it was initially thought that you would need to execute a single-tile walljump in the case that it was the Grey or Blue Keys _(Green area requires Super Flippers, so N/A for that)_, which would've made this trick Merciless as a result. However, it was discovered after the change was made that you can just break the blocks of the 1st Hill in the same manner as the 2nd Hill, which negates the need for the single-tile walljump.